### PR TITLE
MMI-2119-env-name-as-key

### PR DESCRIPTION
**Pull request by bergarces**
## **Description**

Update MMI package dependencies and refactor code to handle the breaking changes.

The changes involve using environment name as the main key to identify a custodian, instead of the api url.

It should not change any existing feature 

## **Related issues**

Fixes: MMI-2119

## **Manual testing steps**

Test MMI build with multiple custodians, including:
- Updating an existing installation and verifying that existing custodian accounts have been migrated correctly
- Adding accounts from a custodian
- Refreshing tokens for custodian accounts

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've clearly explained what problem this PR is solving and how it is solved.
- [X] I've linked related issues
- [X] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/controllers/mmi-controller.js [hunk 1]
```diff
@@ -152,7 +152,11 @@ export default class MMIController extends EventEmitter {
 
         // Trigger this event, listen to sdk, sdk change the state and then Ui is listening for the state changed
         keyring.on(INTERACTIVE_REPLACEMENT_TOKEN_CHANGE_EVENT, (payload) => {
-          log.info(`Interactive refresh token change event for ${payload}`);
+          log.info(
+            `Interactive refresh token change event for ${JSON.stringify(
+              payload,
+            )}`,
+          );
```
Comment (bergarces) [inline app/scripts/controllers/mmi-controller.js:159]: I changed this because it kept being logged as `[Object object]` in the console, which wasn't particularly helpful.

File: ui/pages/institutional/confirm-add-custodian-token/confirm-add-custodian-token.js [hunk 2]
```diff
@@ -75,17 +77,10 @@ const ConfirmAddCustodianToken = () => {
             await dispatch(setProviderType(networkType));
           }
 
-          let custodianName = connectRequest.service.toLowerCase();
-
-          if (connectRequest.service === 'JSONRPC') {
-            custodianName = connectRequest.environment;
```
Comment (bergarces) [inline ui/pages/institutional/confirm-add-custodian-token/confirm-add-custodian-token.js:81]: I'm pretty certain from looking at the config, that in every single case environment is correct, as it seems to always be the name with lowercase.

But we should revert this change if that's not the case.

File: app/scripts/controllers/mmi-controller.js [hunk 3]
```diff
@@ -530,18 +533,19 @@ export default class MMIController extends EventEmitter {
...
 
-    await keyring.replaceRefreshTokenAuthDetails(address, newAuthDetails);
+    await keyring.replaceRefreshTokenAuthDetails(address, refreshToken);
   }
 
   async handleMmiCheckIfTokenIsPresent(req) {
-    const { token, apiUrl } = req.params;
+    const { token, envName } = req.params;
+    // TODO (Bernardo) - Check if this is the case
     const custodyType = 'Custody - JSONRPC'; // Only JSONRPC is supported for now
```
Comment (bergarces) [inline app/scripts/controllers/mmi-controller.js:549]: Is this right? Isn't ECA3 supported as well?

File: ui/helpers/utils/institutional/find-by-custodian-name.ts [hunk 4]
```diff
@@ -13,6 +13,7 @@ type Custodian = {
   version: number;
 };
 
+// TODO (Bernardo) - There can be multiple custodian with the same name, envName should be used instead
```
Comment (bergarces) [inline ui/helpers/utils/institutional/find-by-custodian-name.ts:16]: I haven't changed anything from this function as it's out of scope for this ticket, but I think it would be best to filter by envName/custodianName.

It's being used in two places and both of them have access to the envName.

**Review by shane-t — DISMISSED** [review]

**Review by zone-live — DISMISSED** [review]
Looks good code wise, I'll create some follow up tickets for the @todos you found, and run the extension with this branch to do some manual testing as well.
Awesome work! 🥇

**Review by zone-live — APPROVED** [review]

**Review by zone-live — CHANGES_REQUESTED** [review]
After some manual testing using Qredo, it seems we have a bug. This error happens when connecting MMI to Qredo and trying to import accounts.

Screenshot 2024-02-06 at 17 23 25

Should be related with: https://github.com/consensys-vertical-apps/metamask-institutional/blob/main/packages/custodyKeyring/src/CustodyKeyring.ts#L526

**Review by zone-live — APPROVED** [review]

**Review by shane-t — APPROVED** [review]
